src/main/java/org/cirdles/topsoil/dataset/DatasetResourceFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015 CIRDLES.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.cirdles.topsoil.dataset;
+
+import java.nio.file.Path;
+
+/**
+ *
+ * @author parizotclement
+ */
+public class DatasetResourceFactory {
+
+    private final Path basePath;
+
+    public DatasetResourceFactory(Path basePath) {
+        this.basePath = basePath;
+    }
+
+    public DatasetResource makeDatasetResource(Dataset dataset, Path path) {
+        return new DatasetResource(dataset, basePath, path);
+    }
```

Comment:
You should make a new `DatasetResource` with just a `Dataset`. That way, it can be stored in SQLite, TSV, the cloud or anywhere else. Yes, I do realize that `Dataset`s can be open or closed. That doesn't change anything but implementation.